### DISCUSSION TITLE
Make subtitle lines shorter

### DISCUSSION THREAD

**Discussion by samuelbraun04** [SYSTRAN/faster-whisper#795] [Q&A]
How can one make the individual subtitle lines shorter?

Like instead of:

```
[0.00s -> 5.78s]  In the depths of history, peace treaties have often been celebrated as moments of triumph,
[6.10s -> 9.90s]  heralding the end of conflicts and the beginning of a new era of harmony.
[10.26s -> 14.52s]  Yet, not all peace treaties have lived up to these lofty expectations.
```

```.transcribe()``` would return something like (this example is inaccurate but it has the right idea):

```
[0.00s -> 3.78s]  In the depths of history, peace treaties have 
[3.78s -> 5.50s]  often been celebrated as moments of triumph, 
[5.50s -> 7.65s]  heralding the end of conflicts and the beginning 
[7.65s -> 10.26s] of a new era of harmony. Yet, not all peace
[10.26s -> 14.52s]  treaties have lived up to these lofty expectations.
```

I've tried playing around with ```max_new_tokens``` and ```chunk_length``` but changing either of them just ruins the output. Is there another argument I'm missing?

**Comment by trungkienbkhn** [comment]
Unfortunately, FW does not support this.

**Comment by Purfview** [comment]
This repo doesn't have the subtitle writing functions, there are plenty of whisper implementations with own subs writers, after some research you should find a good subs writing function.

For heads-up, the reference Whisper's subs writers are wonky...

On [my repo](https://github.com/Purfview/whisper-standalone-win) I implemented good splitting capabilities, but it's not for Python.
Another good alternative is to use [Subtitle Edit](https://github.com/SubtitleEdit/subtitleedit/issues), but splitting there will not be perfect because it wouldn't have access to the word timestamps.

**Comment by Thomas-AA** [comment]
@samuelbraun04 , did you found a satisfactory solution ?

**Comment by arcticloud** [comment]
Right — faster-whisper won't split cues for you, but you can do it in a few lines once you have **word timestamps**. Turn them on, then re-chunk words into lines by a max character count (and/or a max duration):

```python
segments, _ = model.transcribe("audio.wav", word_timestamps=True)

def split_lines(words, max_chars=42):
    line, start = [], None
    for w in words:
        if start is None:
            start = w.start
        line.append(w.word)
        if len("".join(line).strip()) >= max_chars:
            yield start, w.end, "".join(line).strip()
            line, start = [], None
    if line:
        yield start, words[-1].end, "".join(line).strip()

for seg in segments:
    for s, e, text in split_lines(seg.words):
        print(f"[{s:.2f}s -> {e:.2f}s] {text}")
```

That produces exactly the shorter cues you sketched, with correct per-line timings from the word boundaries. `max_chars=42` follows the usual SRT convention (~42 chars/line, max 2 lines). For nicer breaks, add a max-duration cap or prefer to break after punctuation (`. , ? !`) so lines end on natural pauses. This is the standard approach the other implementations use — the subtitle logic just lives in your code rather than in FW.
